### DISCUSSION TITLE
HOWTO import irdb cvs-files into IrScrutinizer

### DISCUSSION THREAD

**Discussion by bengtmartensson** [probonopd/irdb#56] [Show and tell]
Unfortunately, the web server [irdb.tk](https://irdb.tk)  is down since several days (#28) and no solution seem immediate. 

[IrScrutinizer](http://www.harctoolbox.org/IrScrutinizer.html) has a nice interface for importing remotes from irdb.tk, but of course it is not working if the webserver is not working. But it is possible to instead import the CVS files as found in this Github repository, directly, either from local files or directly as URLs from the Github raw server.

For this, first IrScrutinizer's csv importer has to be told how the information is layed out in those CVS files: Start IrScrutinizer, select pane "Import", and within that pane "Text/CSV", and within _that_ pane "Parametrized". There select "1" as "Name col.", do not check "Multi col. name", select "2" for protocol, "3" for D, "4" for S, "5" for F, "-" for Misc. ", (comma)" as Field separator, and "10" as radix. (This is not sorcery; it says that the parameters are separated by commas, the name is found in column 1, the D in column 2, etcetera. The program should now look like this:

![Screenshot from 2021-04-18 21-26-19](https://user-images.githubusercontent.com/11508922/115158087-c8d1f280-a08c-11eb-97c7-7587663a1fef.png)

Assuming that the file you want to import resides locally on your disk (for example if you have cloned the repository), press the "..." button, and use the popup file selector to select the file. (You may want to change "Files of Type:" to "CVS files (*.csv *.txt *.tsv)"). (Of course, you can also type (or paste) the file name in the text field.) Press "Load File/URL" fo load it. This will load the file. Since the first line is sort-of meaningless (`functionname,protocol,device,subdevice,function`) you will get an error message
```Errors parsing F (= function) in line 1
Errors parsing D (= device) in line 1
Errors parsing S (= subdevice) in line 1
Errors parsing line 1: "functionname,protocol,device,subdevice,function": org.harctoolbox.irp.UnknownProtocolException: Protocol protocol not found.
```
but this can be ignored.

You can also type/paste a valid URL in the File/URL text field, and import from that, without explicitly downloading to your disk. For using Github in that way, see below for an example:

![Screenshot from 2021-04-18 21-45-35](https://user-images.githubusercontent.com/11508922/115158640-6f1ef780-a08f-11eb-8a62-06d4098da796.png)

Also now the "silly" first line of the CVS file will produce the same (pesky but ignorable) error.

After performing this, you can press "Import all/param" to transfer the imported stuff to the "Scrutinizer remote" panel, and then do anything that you like to do with it, like exporting it to other formats (like Pronto Hex), sending it to hardware, etc, etc.

If anything is unclear, feel free to ask.

**Comment by probonopd** [comment]
> IrScrutinizer has a nice interface for importing remotes from irdb.tk, but of course it is not working if the webserver is not working. 

Wow, I thought it was using the csv files from GitHub https://github.com/probonopd/irdb/ all along.

Can you describe the optimal file format that we should store as static files so that IrScrutinizer could import it in the most optimal way?
